lstchain/reco/reconstructorCC.py

```diff
@@ -227,6 +227,69 @@ def template_interpolation(gain, times, t0, dt, a_hg, a_lg):
     return out
 
 
+@njit(cache=True)
+def template_interpolation_withNSB(empty_waveform, time, is_high_gain, additional_nsb, amplitude, t_0,
+                           t0_template, dt_template, a_hg_template, a_lg_template):
+    """
+    Fast template interpolator using uniformly sampled base with known origin and step to add additional_nsb photons
+    as NSB with a certain amplitude at t_0. The code works for all pixel and selected gains.
+
+    Code interpolation duplicated from function template_interpolation.
+
+    Parameters
+    ----------
+    empty_waveform: empty charge (p.e. / ns) in each pixel and sampled time.
+    time: float64 1D array
+        Times of each waveform samples
+    is_high_gain: boolean 1D array
+        Gain channel used per pixel: True=hg, False=lg
+    additional_nsb: float64 1D array
+        Number of NSB photons to add per pixel
+    amplitude: float 2D array
+        Normalisation factor to apply to the template per photon in each pixel
+    t_0: float 2D array
+        Shift in the origin of time per photon in each pixel
+    t0_template: float64
+        Time of the first value of the pulse templates
+    dt_template: float 64
+        Time step between templates values
+    a_hg_template: float64 1D array
+        Template values for the high gain channel
+    a_lg_template: float64 1D array
+        Template values for the low gain channel
+
+    Returns
+    -------
+    dummy_waveform: float64 2D array
```

Comment:
Either replace dummy_waveform by "nsb_waveform", "empty_waveform" or don't name it here.

Reply:
I renamed `empty_waveform`  to `nsb_waveform`. I adapted the returns docs accordingly.